obfuscation_detection/utils.py

```diff
@@ -162,6 +163,42 @@ def calc_uncommon_instruction_sequences_score(function):
     score = count / sum(function_ngrams.values())
     return score
 
+arithmetic_op = set([HighLevelILOperation.HLIL_ADD, HighLevelILOperation.HLIL_NEG, HighLevelILOperation.HLIL_SUB, HighLevelILOperation.HLIL_MUL, HighLevelILOperation.HLIL_DIVS, HighLevelILOperation.HLIL_MODS])
+boolean_op = set([HighLevelILOperation.HLIL_NOT, HighLevelILOperation.HLIL_AND, HighLevelILOperation.HLIL_OR, HighLevelILOperation.HLIL_XOR])
+arithmetic_counter = 0
+boolean_counter = 0
+
+def traverse_HLIL(il):
+    global arithmetic_counter, boolean_counter
+    if isinstance(il, highlevelil.HighLevelILInstruction):
+        if il.operation in arithmetic_op:
+            arithmetic_counter += 1
+
+        if il.operation in boolean_op:
+            boolean_counter += 1
+
+        for o in il.operands:
+            traverse_HLIL(o)
+
+def calculate_arithmetic_complexity_expressions(function):
+    # check if the hlil AST has been generated for the function
+    if function.analysis_skipped: return 0
```

Comment:
* rename to `calculate_complex_arithmetic_expressions`
* `function.hlil_if_available == None` checks if hlil has been generated and ONLY uses such expressions if it has. contrary, `function.hlil` generates hlil (and makes everything really slow)
* remove global variables due to restructreing
* use `function.hlil_if_available.instructions` instead of `root`. `root` makes iteration harder, since it can also contain hlil blocks (instead of instrucitons). if not careful, one misses instructions.

example:

```python
def calculate_complex_arithmetic_expressions(function):
    # check if the hlil has been generated for the function
    if function.hlil_if_available == None:
        return 0
    # init mba counter
    instr_mba = 0
    # iterate hlil instructions
    for ins in function.hlil_if_available.instructions:
        # if an expression has a boolean and a arithmetic operation, the expression has some arithmetic complexity
        if uses_mixed_boolean_arithmetic(ins):
            instr_mba += 1
    return instr_mba
```